core/src/test/java/utilcalc/core/reportGen/ColdWaterSectionGeneratorTest.java

```diff
@@ -0,0 +1,343 @@
+                        createDateRange("2024-01-01", "2025-01-01"),
+                        createColdWaterSectionInputs(
+                                List.of(waterTariff),
+                                List.of(meterReading1, meterReading2, meterReading3)));
+
+        assertThat(coldWaterSection.name()).isEqualTo("Cold water");
+        assertThat(coldWaterSection.readings().size()).isEqualTo(2);
+        assertThat(coldWaterSection.priceList().size()).isEqualTo(2);
+        assertThat(coldWaterSection.totalAmount()).isEqualTo("4500.00");
+
+        WaterReading waterReading1 = coldWaterSection.readings().getFirst();
+        assertThat(waterReading1.meterId()).isEqualTo("kitchen");
+        assertThat(waterReading1.dateRange())
+                .isEqualTo(createDateRange("2024-01-01", "2024-06-01"));
+        assertThat(waterReading1.startState()).isEqualTo("100");
+        assertThat(waterReading1.endState()).isEqualTo("125");
+        assertThat(waterReading1.consumption()).isEqualTo("25");
+
+        WaterReading waterReading2 = coldWaterSection.readings().getLast();
+        assertThat(waterReading2.meterId()).isEqualTo("kitchen");
+        assertThat(waterReading2.dateRange())
+                .isEqualTo(createDateRange("2024-06-01", "2025-01-01"));
+        assertThat(waterReading2.startState()).isEqualTo("125");
+        assertThat(waterReading2.endState()).isEqualTo("150");
+        assertThat(waterReading2.consumption()).isEqualTo("25");
+    }
+
+    @Test
+    void coldWaterSection_withOneMeterId_withMissingReadings_should_haveCorrectColdWaterReading() {
+        MeterReading meterReading1 = createMeterReading("kitchen", "2024-02-01", "100");
+        MeterReading meterReading2 = createMeterReading("kitchen", "2024-06-01", "125");
+        MeterReading meterReading3 = createMeterReading("kitchen", "2024-09-01", "150");
+
+        WaterTariff waterTariff =
+                createWaterTariff(createDateRange("2024-01-01", "2025-01-01"), "90");
+
+        ColdWaterSection coldWaterSection =
+                generateColdWaterSection(
+                        createDateRange("2024-01-01", "2025-01-01"),
+                        createColdWaterSectionInputs(
+                                List.of(waterTariff),
+                                List.of(meterReading1, meterReading2, meterReading3)));
+
+        assertThat(coldWaterSection.name()).isEqualTo("Cold water");
+        assertThat(coldWaterSection.readings().size()).isEqualTo(4);
+        assertThat(coldWaterSection.priceList().size()).isEqualTo(4);
+        assertThat(coldWaterSection.totalAmount()).isEqualTo("8062.47");
+
+        WaterReading waterReading1 = coldWaterSection.readings().getFirst();
+        assertThat(waterReading1.meterId()).isEqualTo("kitchen");
+        assertThat(waterReading1.dateRange())
+                .isEqualTo(createDateRange("2024-01-01", "2024-02-01"));
+        assertThat(waterReading1.startState()).isEqualTo("93.750");
```

Comment:
Bohuzel si porad nerozumime. Na vstupu musi byt odecty, ktere pokryvaji cele uctovaci obdobi - muzou pokryvat vic, ale ne min. V tomhle pripade odecty nepokryvaji leden 2024, program by mel ohlasit chybu. Pokud by na vstupu byl odecet z 2023-12-01, tak by to bylo ok a program by musel odhadnout stav vodomeru k 2024-01-01.